sycl/source/detail/image_impl.hpp

```diff
@@ -238,7 +238,14 @@ class image_impl final : public SYCLMemObjT {
 
   // Returns the total number of elements in the image
   size_t get_count() const { return size(); }
-  size_t size() const noexcept { return MRange.size(); }
+  size_t size() const {
+    try {
+      return MRange.size();
+    } catch (std::exception &e) {
+      __SYCL_REPORT_EXCEPTION_TO_STREAM("exception in size", e);
```

Comment:
This will be a user-visible message, so it should have more information

Reply:
The exception details comes from the origin (far down the call chain).
This function is to catch it and exit gracefully with whatever propagated from the exception origin.
There is no way at this outermost level can specify the details of the exception.